ratis-common/src/main/java/org/apache/ratis/datastream/impl/DataStreamReplyByteBuffer.java

```diff
@@ -83,6 +85,15 @@ public Builder setDataStreamPacket(DataStreamPacket packet) {
           .setStreamOffset(packet.getStreamOffset());
     }
 
+    public Builder setDataStreamException(DataStreamException exception) {
+      return setSuccess(false)
+          .setBuffer(DataStreamProtoUtils.exception2proto(exception).toByteString().asReadOnlyByteBuffer());
+    }
+
+    public Builder setThrowable(Throwable t) {
```

Comment:
@szetszwo Not sure whether should we wrap DataStreamException in RaftClientReply, and wrap RaftClientReply in DataStreamReplyByteBuffer. Because we wrap StateMachineException in RaftClientReply, if we do not wrap DataStreamException in RaftClientReply, it maybe cause user confused and user maybe hard to understand to the reply.

Reply:
Agree.  Let's wrap DataStreamException in RaftClientReply as your change in #277.  It looks good.